### PR TITLE
Refactor class-based implementation of all the services

### PR DESCRIPTION
**Pull request by gctucker**
Refactor all the services to align their implementation using a class with common members and similar patterns for the main loop etc.  This should help consolidate things as a follow-up with a base class to group the common part and simplify each implementation.

### CODE REVIEW

File: src/send_kcidb.py [hunk 1]
```diff
@@ -24,19 +24,21 @@
...
-    def __init__(self, sub_parser, name):
-        super().__init__(sub_parser, name)
+    def __init__(self, configs, args):
         self._logger = Logger("config/logger.conf", "send_kcidb")
-
-    def __call__(self, configs, args):
         db_config = configs['db_configs'][args.db_config]
         api_token = os.getenv('API_TOKEN')
-        db = kernelci.db.get_db(db_config, api_token)
+        self._db = kernelci.db.get_db(db_config, api_token)
```
Comment (mgalka) [inline src/send_kcidb.py:33]: Further `db` object calls in the script need to be changed to `self._db` i.e. lines: 66, 77, 78, 119 (64, 73, 74, 114)

Comment (gctucker) [reply src/send_kcidb.py:33]: Ah yes, I think I'll just drop the refactoring of `send_kcidb.py` from this PR as I can't really test it.

File: src/timeout.py [hunk 2]
```diff
@@ -130,8 +134,7 @@ class cmd_run(Command):
     ]
 
     def __call__(self, configs, args):
-        timeout = Timeout(configs, args)
-        timeout.run()
+        Timeout(configs, args).run()
```
Comment (JenySadadia) [inline src/timeout.py:137]:
```suggestion
        return Timeout(configs, args).run()
```

Comment (gctucker) [reply src/timeout.py]: That was actually fixed in #175 but I've fixed it here now anyway.

**Review by mgalka — APPROVED** [review]
LGTM

**Review by JenySadadia — CHANGES_REQUESTED** [review]
A suggestion for `RunnerSingleJob._run_single_job`:
```
except Exception:
            self._logger.log_message(logging.ERROR, traceback.format_exc())
            return False
```

**Comment by gctucker** [comment]
> A suggestion for `RunnerSingleJob._run_single_job`:
> 
> ```
> except Exception:
>             self._logger.log_message(logging.ERROR, traceback.format_exc())
>             return False
> ```

Ah yes, this is fixed already in #175 but I'll fix it here as well as it's more correct.

**Review by JenySadadia — APPROVED** [review]

**Pull request merged**